### PR TITLE
Search: link to main project in subproject results

### PR DESCRIPTION
**Pull request by stsewd**
I'm checking for urls stating with `/projects/` only, since we don't want to go back to link to the full URL again...

Closes https://github.com/readthedocs/readthedocs.org/issues/7487

### CODE REVIEW

**Comment by ramirezfranciscof** [comment]
@stsewd @ericholscher Hey, thanks for taking care of this. Would you have an estimate on when you would be able to merge and deploy this?

**Review by ericholscher — APPROVED** [review]
This seems like a good fix for now, sorry for the delay here 👍

File: readthedocs/core/static-src/core/js/doc-embed/search.js [hunk 1]
```diff
@@ -70,6 +70,11 @@ function attach_elastic_search_query_sphinx(data) {
                         }
 
                         var link = result.path + "?highlight=" + $.urlencode(query);
+                        // If we aren't on the main domain of the subproject, link to it.
+                        // TODO: we should always redirect to the main domain instead.
+                        if (result.path.startsWith('/projects/') && !window.location.href.startsWith(result.domain)) {
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/doc-embed/search.js:75]: I think on .com we allow users to customize this path, so we probably need something smarter here in the future, but this should work for 99% of users for now.

Comment (stsewd) [reply readthedocs/core/static-src/core/js/doc-embed/search.js:75]: yeah, I think redirecting to the main project should fix all this, but that's a bigger change in proxito

**Comment by stsewd** [comment]
@ramirezfranciscof this should be live by tomorrow afternoon

**Comment by ltalirz** [comment]
Thanks a lot, guys! I confirm that this has fixed the issue for us.

**Comment by ramirezfranciscof** [comment]
Indeed, all works great now, thanks!

**Pull request merged**